src/AutoRest.CSharp/LowLevel/Output/OperationMethodChainBuilder.cs

```diff
@@ -143,7 +166,7 @@ private ReturnTypeChain BuildReturnTypes()
 
         private ConvenienceMethod BuildConvenienceMethod(ReturnTypeChain returnTypeChain)
         {
-            string name = _protocolBodyParameter == null
+            string name = !_orderedParameters.Where(parameter => parameter.Convenience != KnownParameters.CancellationTokenParameter).ToList().Exists(parameter => !IsParameterTypeSame(parameter.Convenience, parameter.Protocol))
```

Comment:
Better create a function instead of duplicate the code.

Reply:
Actually in this class, there is a lot of such kind of inline operations. If we always to create a function for these operations, I'm worried about it would be hard to read. So I'd prefer Dapeng's suggestion.